package com.learning.shared.exception;


import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestController;

import com.learning.shared.bean.ResponseBean;
import com.learning.shared.constant.StatusCode;
import com.learning.shared.util.JsonUtil;

@RestController
@ControllerAdvice
public class GlobalExceptionHandler {
	
	
	@ExceptionHandler(NullPointerException.class)
	public ResponseBean<String> nullPointerException(NullPointerException exception) {
		log.info("log working");
		exception.printStackTrace();
		ResponseBean<String> responseBean = JsonUtil.createResponseBean(StatusCode.INTERNAL_SERVER_ERROR,"Somthing went wrong please try again!!",null);
		return responseBean;
	}
}
